securesign/rekor
# Add Github action to check upstream dockerfile go image

This pr is related to this slack [message](https://redhat-internal.slack.com/archives/C05M2GGKU7Q/p1693321174551639?thread_ts=1693320909.554809&cid=C05M2GGKU7Q), and involves adding a GitHub action that periodically compares the Images located in the upstream docker files to the ones in our patch files, if they are different the action will generate a patch file to replace our current one, then creates a pr.

## Testing

1. Fork the repo
2. Change settings mentioned in the comments section
3. Modify the sha for the golang image in the redhat/patches/0001-dockerfile.patch file (the sha should still be a valid sha just different from the original) 
5. Run the action and it should create an automated pr  with the new patch file

## Comments
In order for the action to work correctly there are two settings that need to be changed for the repo.

1.  Actions need to be able to create pull requests (settings -> Actions -> General -> Workflow permissions)
6. Actions need read and write permissions (settings -> Actions -> General -> Workflow permissions)
